src/utils/utils.ts

```diff
@@ -1,9 +1,43 @@
 import { Dispatch, SetStateAction } from "react";
-import { querySearchService, urlWithEndSlash } from "../api/browse";
+import { querySearchService } from "../api/browse";
 import { SearchResponseModel } from "../models/dataset";
 import { FacetFilterModel } from "../models/facets";
 
-const CLIENT_URL = new URL(urlWithEndSlash(process.env.REACT_APP_CLIENT_URL!))
+/**
+ * Checks and cleans up URL if last character is or is not forward slash
+ * Duplicate of utils version because of infinite recursion error, needs refactoring
+ * @param url - String of URL
+ * @param endSlash - whether we want to have an end slash or not
+ * @returns URL as specified
+ */
+export const urlWithEndSlash = (url: string) => {
+  const lastCharSlash = url.endsWith("/");
+  return lastCharSlash ? url : url + "/";
+};
+
+const getParsedUrl = (varName: string) => {
```

Comment:
Thanks, that looks better. Actually you could use one-liners for these two functions, i.e. `f = () => x`; instead of `f = () => { return x; }`.

Reply:
d57939a1c88e354507dc0d700ea0e88ea581c40f